pkgs/development/haskell-modules/make-package-set.nix

```diff
@@ -137,9 +137,15 @@ let
     '';
 
   all-cabal-hashes-component = name: version: buildPackages.runCommand "all-cabal-hashes-component-${name}-${version}" {} ''
```

Comment:
Would you be able to add a comment about what `all-cabal-hashes-component` does?  As well as what it returns?

Also, it would be nice to add a comment to the `all-cabal-hashes` argument at the top of this file that it can either be a tarball, or just a directory.

Also, just curious, but how are you making use of this?

Reply:
Sure, will add some comments.

> Also, just curious, but how are you making use of this?

We read a `flake.lock` to get the details about the `all-cabal-hackage` revision we're using (the `commercialhaskell` owned repository), and then we use that to `fetchTarball` and pass it as an `all-cabal-hashes` override. The goal is to use the flake input locking as a mechanism for updating the hackage revision that we use. But that gives us a directory, not a tarball, so it won't work without this patch.